File: app/src/main/java/me/imzack/app/end/view/contract/PlanCreationViewContract.java
package me.imzack.app.end.view.contract;

import me.imzack.app.end.model.bean.FormattedType;
import me.imzack.app.end.view.adapter.TypeGalleryAdapter;

public interface PlanCreationViewContract extends BaseViewContract {

    void showInitialView(TypeGalleryAdapter typeGalleryAdapter, FormattedType formattedType);

    void onContentChanged(boolean isValid);

    void onStarStatusChanged(boolean isStarred);

    void onTypeOfPlanChanged(FormattedType formattedType);

    void onTypeCreationItemClicked();

    void showDeadlinePickerDialog(long defaultDeadline);

    void onDeadlineChanged(CharSequence deadline);

    void showReminderTimePickerDialog(long defaultReminderTime);

    void onReminderTimeChanged(CharSequence reminderTime);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/PlanCreationViewContract.kt
package net.zackzhang.app.end.view.contract

import net.zackzhang.app.end.model.bean.FormattedType
import net.zackzhang.app.end.view.adapter.TypeGalleryAdapter

interface PlanCreationViewContract : BaseViewContract {

    fun showInitialView(typeGalleryAdapter: TypeGalleryAdapter, formattedType: FormattedType)

    fun onContentChanged(isValid: Boolean)

    fun onStarStatusChanged(isStarred: Boolean)

    fun onTypeOfPlanChanged(formattedType: FormattedType)

    fun onTypeCreationItemClicked()

    fun showDeadlinePickerDialog(defaultDeadline: Long)

    fun onDeadlineChanged(deadline: CharSequence)

    fun showReminderTimePickerDialog(defaultReminderTime: Long)

    fun onReminderTimeChanged(reminderTime: CharSequence)
}
